Ladybird/AppKit/Application/EventLoopImplementation.mm

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2023, Tim Flynn <trflynn89@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Assertions.h>
+#include <AK/IDAllocator.h>
+#include <LibCore/Event.h>
+#include <LibCore/Notifier.h>
+#include <LibCore/ThreadEventQueue.h>
+
+#import <Application/EventLoopImplementation.h>
+#import <System/Cocoa.h>
+#import <System/CoreFoundation.h>
+
+namespace Ladybird {
+
+struct ThreadData;
+static thread_local ThreadData* s_thread_data;
+
+struct ThreadData {
+    static ThreadData& the()
+    {
+        if (!s_thread_data) {
+            // FIXME: Don't leak this.
+            s_thread_data = new ThreadData;
+        }
+
+        return *s_thread_data;
+    }
```

Comment:
I believe on master, Zaggy changed this to just 

```
struct ThreadData {
   static ThreadData& the() {
       static thread_local ThreadData s_thread_data;
       return s_thread_data;
   }
}
```
in the LibCore/Unix event loop impl to avoid the leak here.